torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -937,9 +937,13 @@ Tensor TensorExprKernel::bindInput(const torch::jit::Value* input) {
         bufferArgs_.emplace_back(inBuffer);
         break;
       }
+      ExprHandle flat_size = 0;
+      for (size_t i = 0; i < *tt->sizes().size(); i++) {
+        flat_size = flat_size + *tt->sizes()[i] * *tt->strides()[i];
```

Comment:
should this actually be `max(flat_size, sizes[i] * strides[i])` ? e.g. a 2x4 tensor with strides [4, 1] has size 8 (2\*4), not 12 (2\*4 + 4\*1)

Reply:
I think technically it should be `sum((sizes[i] - 1)*strides[i]) + 1` - on every dimension we can attempt to access element `sizes[i]-1` and as we're accessing it we use `strides[i]`. In [2,4]-[4,1] example this would yield `1*4 + 3*1 + 1 = 8`.

Reply:
@ZolotukhinM is right about this! Updated.